### DISCUSSION TITLE
KeyError when I try to connect the Client

### DISCUSSION THREAD

**Discussion by akarri2001** [ramnes/notion-sdk-py#23] [Q&A] [answered]
Hi! First, I would like to thank you for creating this library. I had a problem when I was trying to follow along with the README and I keep receiving a "KeyError" when I try to connect the notion Client. 

I've included an image below. The token is for some test integration I made just for this purpose so I don't mind if anyone wants to experiment with it. I tried the code on Google Colab to ensure that the problem wasn't just my computer. 

Any help would be appreciated! 

![error](https://user-images.githubusercontent.com/835072/118371638-d1471b80-b5ad-11eb-8373-c0185e94bfa9.png)

*Edit by ramnes: I modified the image to mask the secret token.*

**Comment by Midnitte** [comment, marked as answer]
I believe you need TOKEN to be in quotations

```import os
from notion_client import Client
from notion_client.api_endpoints import DatabasesEndpoint
os.environ['NOTION_TOKEN'] = "secret_token_here"
notion = Client(auth=os.environ["NOTION_TOKEN"])

from pprint import pprint

list_of_users = notion.users.list()
pprint(list_of_users.json())```

**Reply by akarri2001** [reply]
Thank you!!! This helped out a lot!

**Comment by ramnes** [comment]
The idea is that you shouldn't hardcode a token, but use environment variables instead. You can read more on this here: https://12factor.net/config

In your situation, that means removing the `TOKEN = ...` line, adding quotes for `os.environ["TOKEN"]`, and running your Python script with a TOKEN environment variable. If you're using Linux or macOS, you can do this with `TOKEN=secret_... python myscript.py`.

**Reply by akarri2001** [reply]
Thank you! I learned something new today :)

**Comment by aahnik** [comment]
The best way to store secrets is a `.env` file. This convention is used across multiple programming languages.

**Create a `.env` file and store your variables like this.**

```py
VARIABLE=val
NOTION_TOKEN=secret_abcdefgh1234
````

**In your python code, you need to load the variables defined in the `.env` file.** 

```python
# settings.py

import os

from dotenv import load_dotenv # pip install python-dotenv

load_dotenv()

VARIABLE = os.getenv("VARIABLE")
NOTION_TOKEN = os.getenv("NOTION_TOKEN")

```

**You can import and use these variables in your python module where you actually need them.**

```python
# script.py

from notion_client import Client

import settings

# in this scope you can use the variables defined in settings.py

print(settings.VARIABLE) # don't print secrets

notion = Client(auth=settings.NOTION_TOKEN)

# ...
```
